src/test/java/teammates/test/pageobjects/AppPage.java

```diff
@@ -903,6 +918,20 @@ public AppPage verifyHtmlMainContent(String filePath) throws IOException {
         return verifyHtmlPart(MAIN_CONTENT, filePath);
     }
 
+    public AppPage verifyHtmlMainContentWithReloadRetry(final String filePath) throws IOException {
+        return RetryManager.runUntilNoException(new RetryableTaskReturnsThrows<AppPage, IOException>("HTML verification") {
+            @Override
+            public AppPage run() throws IOException {
```

Comment:
What I means is the generic parameter should correspond to the exception that is going to be thrown. This makes more sense. But what if there are multiples exceptions that are going to be thrown?